ensenso_camera/src/camera.cpp

```diff
@@ -282,27 +282,22 @@ void Camera::updateGlobalLink(ros::Time time, std::string frame, bool useCachedT
   cameraNode[itmLink][itmTarget] = TARGET_FRAME_LINK + "_" + serial;
 
   // Update the transformation to the target frame in the NxLib according to
-  // the current information from TF. Only if the link frame and target frame differs. Otherwise the transformation will
-  // be doubled. If they're the same, the global Link must be the identity transform and the tf will be in the camera
-  // link. (camera -> link in Nxlib or link -> camera in ROS).
-  if (frame != linkFrame)
+  // the current information from TF. Only if the link frame and target frame differs.
+  geometry_msgs::TransformStamped transform;
+  if (useCachedTransformation && transformationCache.count(frame) != 0)
   {
-    geometry_msgs::TransformStamped transform;
-    if (useCachedTransformation && transformationCache.count(frame) != 0)
-    {
-      transform = transformationCache[frame];
-    }
-    else
-    {
-      transform = tfBuffer.lookupTransform(linkFrame, frame, time, ros::Duration(TRANSFORMATION_REQUEST_TIMEOUT));
-      transformationCache[frame] = transform;
-    }
-    tf2::Transform tfTrafo;
-    tf2::convert(transform.transform, tfTrafo);
-    NxLibItem()[itmLinks][TARGET_FRAME_LINK + "_" + serial].setNull();
-    NxLibItem()[itmLinks].setNull();
-    writePoseToNxLib(tfTrafo, NxLibItem()[itmLinks][TARGET_FRAME_LINK + "_" + serial]);
+    transform = transformationCache[frame];
+  }
+  else
+  {
+    transform = tfBuffer.lookupTransform(linkFrame, frame, time, ros::Duration(TRANSFORMATION_REQUEST_TIMEOUT));
+    transformationCache[frame] = transform;
   }
+  tf2::Transform tfTrafo;
+  tf2::convert(transform.transform, tfTrafo);
+  NxLibItem()[itmLinks][TARGET_FRAME_LINK + "_" + serial].setNull();
+  NxLibItem()[itmLinks].setNull();
+  writePoseToNxLib(tfTrafo, NxLibItem()[itmLinks][TARGET_FRAME_LINK + "_" + serial]);
```

Comment:
Note that we can safely remove the `if (frame != linkFrame)` since that is now handled above. The large changeset is indentation, and I removed what I believe to be a no longer applicable comment.